### PR TITLE
Add unit tests for the utils package

### PR DESCRIPTION
**Pull request by albertoleal**
This PR doesn't add any new behaviour, but adds unit tests.

cc: @cloudfoundry/cf-garden

### CODE REVIEW

File: libcontainer/utils/utils_test.go [hunk 1]
```diff
@@ -23,3 +29,96 @@ func TestGenerateName(t *testing.T) {
...
+
+	path = CleanPath("rootfs")
+	if path != "rootfs" {
+		t.Errorf("expected to received 'rootfs' and received %s", path)
+	}
+
+	path = CleanPath("../../../var")
+	if path != "var" {
+		t.Errorf("expected to received 'var' and received %s", path)
+	}
```
Comment (cyphar) [inline libcontainer/utils/utils_test.go]: Please add a test for `/../../../var`. It's one of the more important edge cases in path code.

**Comment by albertoleal** [comment]
@cyphar Done!

File: libcontainer/utils/utils_test.go [hunk 2]
```diff
@@ -23,3 +29,101 @@ func TestGenerateName(t *testing.T) {
...
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if path != fmt.Sprintf("%s/%s", pwd, "rootfs") {
+		t.Errorf("expected rootfs to be abs and was %s", path)
+	}
+}
+
+func TestResolveRootfsWithSymlinc(t *testing.T) {
```
Comment (mrunalp) [inline libcontainer/utils/utils_test.go]: typo: TestResolveRootfsWithSymlinc -> TestResolveRootfsWithSymlink ?

File: libcontainer/utils/utils_test.go [hunk 3]
```diff
@@ -23,3 +29,101 @@ func TestGenerateName(t *testing.T) {
...
+	dir := "rootfs"
+	os.Symlink("/var", dir)
```
Comment (cyphar) [inline libcontainer/utils/utils_test.go]: This test assumes that we have to be root to run the test suite, right? Why not set up some directory in `/tmp` where you make the symlink.

**Comment by albertoleal** [comment]
- Using tmp directory so we do not need to run the tests as `root`.
- Conflicts solved.

**Comment by cyphar** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM
